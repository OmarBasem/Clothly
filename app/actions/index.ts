import {fetchUserAction} from "./user";
import {fetchProductsAction} from "./products";

export {fetchUserAction, fetchProductsAction}