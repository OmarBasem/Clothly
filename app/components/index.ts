import Title from "./Title";
import Back from './Back';
import TopBar from "./TopBar";
import SearchBar from "./SearchBar";
import Product from "./Product";
import ProductsRow from "./ProductsRow";


export {
    Title,
    Back,
    SearchBar,
    TopBar,
    Product,
    ProductsRow
}