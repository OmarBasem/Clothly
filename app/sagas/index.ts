import {all} from 'redux-saga/effects';

import user from './user'
import products from "./products";

const root = function* root() {
    yield all([
        user(),
        products()
    ]);
};

export default root;