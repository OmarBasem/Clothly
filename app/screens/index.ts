import HomeScreen from "./Home/HomeScreen";
import ProductScreen from "./Home/ProductScreen";
import CartScreen from "./Cart/CartScreen";
import ProfileScreen from "./Profile/ProfileScreen";
import WishlistScreen from "./Wishlist/WishlistScreen";

export {
    HomeScreen,
    ProductScreen,
    CartScreen,
    ProfileScreen,
    WishlistScreen
}