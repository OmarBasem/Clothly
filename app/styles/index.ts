import {DefaultTheme} from "@react-navigation/native";

export const theme = {
    mainColor: '#1C284F'
}


export const navTheme = {
    ...DefaultTheme,
    colors: {
        ...DefaultTheme.colors,
        background: '#fff'
    },
};
