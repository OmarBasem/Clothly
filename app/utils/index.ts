import {Dimensions} from "react-native";

export const sliceArrIntoRows = (data: Array<any>, n: number) => {
    const result = [];
    let temp = [];
    for (let i = 0; i < data.length; ++i) {
        if (i > 0 && i % n === 0) {
            result.push(temp);
            temp = [];
        }
        temp.push(data[i]);
    }
    if (temp.length > 0) {
        while (temp.length !== n) {
            temp.push(null);
        }
        result.push(temp);
    }
    return result;
};

export const W = (percentage: number) => Dimensions.get('window').width / 100 * percentage;
export const H = (percentage: number) => Dimensions.get('window').height / 100 * percentage;