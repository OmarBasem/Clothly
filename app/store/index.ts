import {configureStore} from '@reduxjs/toolkit'
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist'
import rootReducer from '../reducers'
import AsyncStorage from '@react-native-async-storage/async-storage';
import createSagaMiddleware from 'redux-saga'
import sagas from "../sagas";

const persistConfig = {
    key: 'root',
    storage: AsyncStorage,
    whitelist: [] // persist the user data in the redux store
}

const persistedReducer = persistReducer(persistConfig, rootReducer)
const sagaMiddleware = createSagaMiddleware()

export const store = configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
            },
        }).concat(sagaMiddleware),
})
sagaMiddleware.run(sagas)

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export const persistor = persistStore(store)


