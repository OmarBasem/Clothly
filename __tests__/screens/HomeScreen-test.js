import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store'
import renderer from "react-test-renderer";
import {HomeScreen} from '../../app/screens'

const testProduct = require('../data/testProduct.json')

const productsMockState = {
    categories: {MEN: {"0": testProduct}},
    filteredProducts: null,
    currentCategory: 'MEN',
    didFetch: {MEN: true},
    selectedProduct: "0"
};

describe('Testing Screens rendering', () => {

    test('HomeScreen renders correctly', () => {
        const mockStore = configureStore()
        const store = mockStore({products: productsMockState})
        const tree = renderer.create(<Provider store={store}><HomeScreen/></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})