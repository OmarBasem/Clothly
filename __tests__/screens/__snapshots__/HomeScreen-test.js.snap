// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Screens rendering HomeScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="HomeScreen"
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": "96%",
      }
    }
    testID="TopBar"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="MEN"
    >
      <Text
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            Object {
              "color": "#1C284F",
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        MEN
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="WOMEN"
    >
      <Text
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        WOMEN
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="KIDS"
    >
      <Text
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        KIDS
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="FTW"
    >
      <Text
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        FTW
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="ACCESSORIES"
    >
      <Text
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            Object {},
          ]
        }
      >
        ACCESSORIES
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="SEARCH"
      selectionColor="#1C284F"
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "grey",
          "borderRadius": 8,
          "borderWidth": 1,
          "fontSize": 16,
          "height": 40,
          "marginBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 32,
          "width": 720,
        }
      }
      testID="SearchBar"
      value=""
    />
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "grey",
            "fontSize": 20,
          },
          Object {
            "position": "absolute",
            "right": 8,
            "top": 8,
            "zIndex": 1,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï”…
    </Text>
  </View>
  <RCTScrollView
    data={
      Array [
        Array [
          Object {
            "colors": Array [
              "#C73C3C",
              "#262626",
              "#C1CFD0",
              "#FFFFFF",
              "#1E4B4A",
            ],
            "description": "Opt for a bold look with this T-shirt. With its iconic Tiger print on the chest, it is sure to turn heads. Pair it with jogging trousers and sneakers for a casual sportswear look.
Short-slleved T-shirt.
Round neck.
Tiger embroidered on the chest.",
            "id": 0,
            "imageURL": "https://www.juniorcouture.ae/dw/image/v2/BGHV_PRD/on/demandware.static/-/Sites-master-catalog/default/dwdc8636f2/images/10014476/KEN-K25100_1.jpg?sw=800&sh=800",
            "more": Array [
              1,
              2,
              3,
              4,
            ],
            "price": 250,
            "sizes": Array [
              "S",
              "M",
              "X",
              "XL",
            ],
            "title": "Tiger Medium Red",
          },
          null,
        ],
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="MEN-clothing"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "width": 750,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderColor": "lightgrey",
                "borderWidth": 1,
                "marginBottom": 8,
                "marginLeft": 12,
                "opacity": 1,
                "padding": 8,
                "width": 337.5,
              }
            }
            testID="item-0-0"
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "height": 450,
                    "width": 300,
                  },
                ]
              }
            >
              <FastImageView
                resizeMode="contain"
                source={
                  Object {
                    "uri": "https://www.juniorcouture.ae/dw/image/v2/BGHV_PRD/on/demandware.static/-/Sites-master-catalog/default/dwdc8636f2/images/10014476/KEN-K25100_1.jpg?sw=800&sh=800",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
              testID="title-0-0"
            >
              Tiger Medium Red
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              AED 
              250
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
